forbidigo/patterns.go

```diff
@@ -0,0 +1,43 @@
+package forbidigo
+
+import (
+	"fmt"
+	"regexp"
+	"regexp/syntax"
+	"strings"
+)
+
+type pattern struct {
+	pattern *regexp.Regexp
+	msg     string
+}
+
+func parse(ptrn string) (*pattern, error) {
+	p := &pattern{}
```

Comment:
just a style thing but my preference here would be to avoid dot notation and construct the final pattern at the end before we return it

Reply:
I usually lean that way too (not mutating structs if it can be avoided), not sure why I went the other way here. Probably. a hangover from the first iteration of this PR.